xarray/core/variable.py

```diff
@@ -119,6 +119,11 @@ def as_variable(obj, name=None) -> "Union[Variable, IndexVariable]":
     if isinstance(obj, Variable):
         obj = obj.copy(deep=False)
     elif isinstance(obj, tuple):
+        if isinstance(obj[1], DataArray):
+            raise DeprecationWarning(  # TODO: change into TypeError
+                "Using a DataArray object to construct a variable is "
+                "ambiguous, please extract the data using the .data property"
+            )
```

Comment:
I think it would be good to state when the warning is turned into a `TypeError` (something like "From 0.19.0 on this will raise a TypeError")